src/guide/migration/emits-option.md

```diff
@@ -0,0 +1,62 @@
+---
+title: emits Option
+badges:
+  - new
+---
+
+# `emits` Option <MigrationBadges :badges="$frontmatter.badges" />
+
+## Overview
+
+Vue 3 now offers an `emits` option similar to the existing `props` option. This option can be used to define the events that a component can emit to its parent.
+
+## 2.x Behavior
+
+In Vue 2, you can define the props that a component received, but you can't define which ebents it can emit:
+
+```javascript
+<template>
+  <p>{{ text }}</p>
+  <button v-on:click="$emit('accepted')">OK</button>
+</template>
+<script>
+  export default {
+    props: ['text'],
+  }
+</script>
+```
+
+## 3.x Behavior
+
+Similar to props, the events that the component emit can now be defined with the `emits` option.
+
+```javascript
+<template>
+  <p>{{ text }}</p>
+  <button v-on:click="$emit('accepted')">OK</button>
+</template>
+<script>
+  export default {
+    props: ['text'],
+    emits: ['accepted']
+  }
+</script>
+```
+
+The option also excepts an object notation, which allows to define validators for the arguments that have passed with the emitted event, similar to validators in props definitions.
+
+For more Information on this, please read the [API documentation for this feature](../api/options-data.md#emits).
+
+## Migration Strategy
+
+Even though this is the new optional feature, it is highly recommended that you document all of the emitted events of your components this way because of the [removal of the `.native` modifier](./native-modifier-removed.md).
+
+All events not defined with `emits` are now added as DOM event listeners to the components root node (unless `inheritAttrs: false` has been set).
```

Comment:
Is it worth explicitly mentioning the 'event proxying' pattern from Vue 2? e.g. Where a button component would listen for a `click` event on the `<button>` element and then re-emit it? In Vue 3 this will now cause two `click` events to be fired, one for the native `click` and one from the component. `emits` fixes the problem, though in practice it is simpler just to use the native event and remove the proxying.

Not sure if this is the right place to mention it as `emits` isn't actually required to solve it. Perhaps it makes more sense to be on the **'`$listeners` removed'** page?

Reply:
I think that example does fit the topic of this page here.

I'll try and come up with something.

Reply:
Added an example, please check it out @skirtles-code